internal/cli/action.go

```diff
@@ -0,0 +1,59 @@
+package cli
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/spf13/cobra"
+)
+
+func newActionCmd() *cobra.Command {
+	result := &cobra.Command{
+		Use:   "action",
+		Short: "perform the sepcified action",
+		Long: `Perform the specified action via a call to Tilt's API server.
+`,
+	}
+
+	result.AddCommand(newTriggerCmd())
+
+	return result
+}
+
+func newTriggerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "trigger [RESOURCE_NAME]",
+		Short: "trigger an update for the specified resource",
+		Long: `Trigger an update for the specified resource.
+
+If no file changes for the resource are pending, this will force a full rebuild.
+
+If it is a manual resource with pending file changes, this will cause a build of those pending changes.
+`,
+		Args: cobra.ExactArgs(1),
+		Run:  triggerUpdate,
+	}
+	cmd.Flags().IntVar(&webPort, "port", DefaultWebPort, "Port for the Tilt HTTP server")
+	return cmd
+}
+
+func triggerUpdate(cmd *cobra.Command, args []string) {
+	resource := args[0]
+	url := fmt.Sprintf("http://localhost:%d/api/trigger", webPort)
+	payload := []byte(fmt.Sprintf(`{"manifest_names":[%q]}`, resource))
+	res, err := http.Post(url, "application/json", bytes.NewBuffer(payload))
```

Comment:
feel free to ignore this for this PR, but we should probably have a function for making an API request from a tilt command so that we're not re-figuring out/implementing error handling and messaging each time and it's consistent across commands. (at least `dump` and `args` also do this)

Reply:
yeahhhhh not a bad idea.

also whoops I forgot that I was gonna actually use the `payload` struct...